src/main/java/org/opentripplanner/transit/raptor/rangeraptor/standard/stoparrivals/Stops.java

```diff
@@ -31,38 +32,20 @@ public Stops(
      * Setup egress arrivals with a callback which is notified when a new transit egress arrival happens.
      */
     public void setupEgressStopStates(
-            Iterable<RaptorTransfer> egressPaths,
-            Consumer<EgressStopArrivalState<T>> transitArrivalCallback
+            EgressPaths egressPaths,
+            DestinationArrivalListener destinationArrivalListener
     ) {
-        for (int round = 1; round < stops.length; round++) {
-            for (RaptorTransfer egressPath : egressPaths) {
-                if(stops[round][egressPath.stop()] == null) {
-                    EgressStopArrivalState<T> state = new EgressStopArrivalState<>(
-                        round,
-                        egressPath,
-                        transitArrivalCallback
-                    );
-                    stops[round][egressPath.stop()] = state;
-                }
-                else {
-                    throw new IllegalStateException(""
-                        + "Currently Raptor do not support multiple access/egress paths to the "
-                        + "same stop. If this exception occurs and OTP was serving a normal "
-                        + "use-case, then this needs to be fixed. For example this needs to be "
-                        + "fixed if OTP should support more than on access/egress mode. "
-                        + "See issue #3300. Details: "
-                        + "State exist for stop: " + egressPath.stop() + ", round: " + round
-                    );
-                }
-            }
+        for (int i = 1; i < stops.length; i++) {
+            final int round = i;
```

Comment:
~Why do we loop over all stops and use it as the round?~ The array name makes this slightly unintuitive.

Reply:
This "stops" name is old legacy from the R5 implementation, I changes all of them I could find to "arrivals/stopArrivals".